internal/app/backend/backend_service.go

```diff
@@ -65,6 +65,17 @@ func newBackend(cfg config.View) (*backendService, error) {
 // proposals which are then evaluated to generate results. FetchMatches method
 // streams these results back to the caller.
 func (s *backendService) FetchMatches(req *pb.FetchMatchesRequest, stream pb.Backend_FetchMatchesServer) error {
+	// Validate that the function configuration and match profiles are provided.
+	if req.Config == nil {
+		logger.Error("invalid argument - match function config cannot be nil")
+		return status.Errorf(codes.InvalidArgument, "match function config cannot be nil")
```

Comment:
This wording makes sense for Go callers, but not from callers in other languages.  Please word it in a way which is more natural to protobufs and rest callers.  Eg, "FetchMatchesRequest requires config to be set."

Reply:
Agreed that nil was a golang specific reference. Removed it. User visible errors need to be user friendly though - so i dont think protobuf specific references should be in them either. Replaced with a generic user friendly string.